content/posts/how-to-evolve-application-cicd/index.md

```diff
@@ -0,0 +1,336 @@
++++
+title = "How to evolve an application to make it more robust and maintainable - Setting up the CICD"
+description = ""
+date = 2023-01-15
+featured = false
+draft = false
+comment = true
+toc = true
+reward = true
+pinned = false
+categories = [
+"quality", "test", "CICD"
+]
+tags = [
+""
+]
+series = [
+" How to evolve an application to make it more robust and maintainable"
+]
+images = []
++++
+
+<!--more-->
+
+
+## Foreword
+
+In the last article we detailed the tests added to our project. 
+These tests are useful to check our development locally as we use TDD. But they are also useful to check if there are regressions.
+In this post we will : 
+
+* define what is the CICD
+* How to add the CICD settings in our project
+
+## What is the CICD
+
+The CICD defines Continuous Integration Continuous Delivery. 
+The CICDCD  is Continuous Integration Continuous Delivery Continuous Deployment
+
+When we talk on CICD ( or CICDCD) we often also referred to as the CICD pipelines.
+CICD pipeline defines a process where certain steps are executed one after the other, all with the aim of verifying the application before delivering it and deploying it.
```

Comment:
with the aim of building, and verifying the application before delivering it, and deploying it

Reply:
Done